python/src/spark_rapids_ml/umap.py

```diff
@@ -514,14 +654,14 @@ def __init__(
         self.raw_data_ = raw_data_
 
     @property
-    def embedding(self) -> List[List[float]]:
-        if isinstance(self.embedding_, list):
+    def embedding(self) -> np.ndarray:
+        if isinstance(self.embedding_, np.ndarray):
             return self.embedding_
         return self.embedding_.value
 
     @property
-    def raw_data(self) -> List[List[float]]:
-        if isinstance(self.raw_data_, list):
+    def raw_data(self) -> np.ndarray:
+        if isinstance(self.raw_data_, np.ndarray):
```

Comment:
Pyspark properties generally return python base types.  Wondering if these public APIs should convert to base types here (for external users), while the internal code can still reference the underlying numpy variables, e.g. `self.raw_data_`.  Not sure if others have an opinion...

Reply:
In that case, probably would be good return it list-converted without changing the internal references.